pkg/apis/core/helper/cloudprofile.go

```diff
@@ -32,10 +32,13 @@ func FindMachineImageVersion(machineImages []core.MachineImage, name, version st
 }
 
 // DetermineLatestMachineImageVersions determines the latest versions (semVer) of the given machine images from a slice of machine images
-func DetermineLatestMachineImageVersions(images []core.MachineImage) (map[string]core.MachineImageVersion, error) {
+func DetermineLatestMachineImageVersions(images []core.MachineImage, allowEmptyVersions bool) (map[string]core.MachineImageVersion, error) {
```

Comment:
`allowEmptyVersions` can be dropped from this function as it can deal with empty versions already, right?

Reply:
For empty image versions with `allowEmptyVersions = false` it would run into the subsequent error. I will remove it nevertheless, as empty versions are checked at another point anyway.